src/app/(pages)/profile/edit/_components/profile-form.tsx

```diff
@@ -49,40 +50,49 @@ interface ProfilePageProps {
 export default function ProfileForm({ userInfo }: ProfilePageProps) {
     const queryClient = useQueryClient()
     const router = useRouter()
-    const { setBottomBarContent } = useAppStore(s => ({
+    const { user, ready } = usePrivy()
+
+    const { setBottomBarContent, isPageTransitioning } = useAppStore(s => ({
         setBottomBarContent: s.setBottomBarContent,
+        isPageTransitioning: s.isPageTransitioning,
     }))
     const [avatarChanged, setAvatarChanged] = useState(false)
 
     const [state, formAction] = useFormState(validateProfileAction, initialState)
 
-    useLayoutEffect(() => {
-        setBottomBarContent(
-            <Button
-                type='submit'
-                form='profile-form'
-                className='mb-3 h-[46px] w-full rounded-[2rem] bg-cta px-6 py-[11px] text-center text-base font-semibold leading-normal text-white shadow-button active:shadow-button-push'>
-                Save
-            </Button>,
-        )
+    useEffect(() => {
+        if (!isPageTransitioning) {
+            setBottomBarContent(
+                <Button
+                    type='submit'
+                    form='profile-form'
+                    className='mb-3 h-[46px] w-full rounded-[2rem] bg-cta px-6 py-[11px] text-center text-base font-semibold leading-normal text-white shadow-button active:shadow-button-push'>
+                    Save
+                </Button>,
+            )
+        }
 
         return () => {
             setBottomBarContent(null)
         }
-    }, [setBottomBarContent])
+    }, [setBottomBarContent, isPageTransitioning])
 
     useEffect(() => {
+        if (!ready || user?.id == null) {
+            toast('User state not ready. Please try again.')
+            return
+        }
         if (state?.message) {
             toast(state.message)
         }
 
         if (state?.message === 'Profile updated successfully') {
             queryClient.invalidateQueries({
-                queryKey: ['userInfo'],
+                queryKey: ['user-info', user?.id],
```

Comment:
It wasn't updating the profile information immediately with the previous query key. When I check the file use-user-info.ts on the main branch, it shows

```
export const useUserInfo = () => {
    const { user } = usePrivy()

    return useQuery({
        queryKey: ['user-info', user?.id],
        queryFn: async () => await fetchUserInfo(user?.id),
        enabled: Boolean(user?.id),
        select: data => {
            const address = user?.wallet?.address as Address | undefined
            const avatarFallback = address && blo(address)
            return {
                avatar: data?.avatar ?? avatarFallback,
                displayName: data?.displayName,
                address,
            }
        },
    })
}
```

The query key is ['user-info', user?.id] unless you're referring to another query or it was changed.

Reply:
I see, my bad, I accidentally confused it since I am migrating to camelCase notations, to follow how wagmi does it, and you can notice that I already did with some like userIsAdmin. But you were absolutely right about this one, sorry about the mistake! 😅